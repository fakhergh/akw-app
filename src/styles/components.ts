import { Components, Theme } from '@mui/material/styles';

export const components: Components<Omit<Theme, 'components'>> = {
    MuiAppBar: {
        styleOverrides: {
            root: ({ theme }) => ({
                background: 'unset',
                width: `calc(100% - ${theme.spacing(4)})`,
                margin: '1rem',
                backgroundColor: theme.palette.background.paper,
                boxShadow: theme.shadows[1],
                color: theme.palette.text.primary,
                zIndex: theme.zIndex.drawer + 1,
            }),
        },
    },
    MuiDrawer: {
        styleOverrides: {
            root: ({ theme }) => ({
                borderRadius: '1rem',
                flexShrink: 0,
                whiteSpace: 'nowrap',
                boxSizing: 'border-box',
                '& .MuiPaper-root': {
                    boxShadow: theme.shadows[1],
                    background: theme.palette.background,
                    height: 'calc(100vh - 2rem)',
                },
            }),
        },
    },
    MuiPaper: {
        styleOverrides: {
            root: ({ theme }) => ({
                elevation: theme.shadows[0],
                borderRadius: theme.spacing(2),
                background: 'unset',
                backgroundColor: theme.palette.background.paper,
            }),
        },
    },
    MuiOutlinedInput: {
        styleOverrides: {
            root: ({ theme }) => ({
                borderRadius: theme.spacing(2),
            }),
        },
    },
    MuiDialogContent: {
        styleOverrides: {
            root: {
                paddingTop: '1rem !important',
            },
        },
    },
    MuiTableCell: {
        styleOverrides: {
            root: ({ theme }) => ({
                borderColor: theme.palette.divider,
            }),
        },
    },
    MuiButton: {
        styleOverrides: {
            root: ({ theme }) => ({
                borderRadius: theme.spacing(2),
            }),
            contained: ({ theme }) => ({
                boxShadow: theme.shadows[1],
            }),
        },
    },
};
