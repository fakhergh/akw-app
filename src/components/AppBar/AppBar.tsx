import IconMenu from '@mui/icons-material/Menu';
import IconMenuOpen from '@mui/icons-material/MenuOpen';
import MuiAppBar, { AppBarProps as MuiAppBarProps } from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import { styled } from '@mui/material/styles';
import Toolbar from '@mui/material/Toolbar';

export interface AppBarProps extends MuiAppBarProps {
    display?: 'block' | 'none';
    open?: boolean;
    drawerWidth: number | string;
    onMenuButtonClick?: () => void;
}

const BaseAppBar = styled(MuiAppBar, {
    shouldForwardProp: (prop) => prop !== 'open' && prop !== 'drawerWidth',
})<Omit<AppBarProps, 'drawerWidth'> & { drawerWidth: string | number }>(
    ({ theme, open, display, drawerWidth }) => ({
        display,
        transition: `
            width ${theme.transitions.duration.leavingScreen}ms ${theme.transitions.easing.sharp},
            margin ${theme.transitions.duration.leavingScreen}ms ${theme.transitions.easing.sharp},
                background 500ms ${theme.transitions.easing.sharp}
    `,
        ...(open && {
            marginLeft: drawerWidth,
            width: `calc(100% - ${theme.spacing(6)} - ${drawerWidth})`,
            transition: `
                width ${theme.transitions.duration.enteringScreen}ms ${theme.transitions.easing.sharp},
                margin ${theme.transitions.duration.enteringScreen}ms ${theme.transitions.easing.sharp},
                background 500ms ${theme.transitions.easing.sharp}
    `,
        }),
    }),
);

const AppLogoWrapper = styled(Box, {
    shouldForwardProp: (prop) => prop !== 'collapsed',
})<{ collapsed?: boolean }>(({ theme, collapsed }) => ({
    transform: `scale(${collapsed ? 1 : 0})`,
    transition: theme.transitions.create('transform', {
        delay: collapsed ? theme.transitions.duration.short : 0,
        easing: theme.transitions.easing.sharp,
        duration: collapsed
            ? theme.transitions.duration.enteringScreen
            : theme.transitions.duration.short,
    }),
}));

export function AppBar({
    children,
    onMenuButtonClick,
    drawerWidth,
    open,
    ...props
}: AppBarProps) {
    return (
        <BaseAppBar {...props} open={open} drawerWidth={drawerWidth}>
            <Toolbar>
                <Box display="flex" alignItems="center">
                    <IconButton
                        color="inherit"
                        onClick={onMenuButtonClick}
                        edge={false}
                        aria-label="btn-menu"
                    >
                        {open ? <IconMenuOpen /> : <IconMenu />}
                    </IconButton>
                    <AppLogoWrapper
                        collapsed={!open}
                        display="flex"
                        alignItems="center"
                    />
                </Box>
                <Box sx={{ flexGrow: 1 }} />
                {children}
            </Toolbar>
        </BaseAppBar>
    );
}
