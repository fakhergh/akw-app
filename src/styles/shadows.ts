import { Shadows } from '@mui/material/styles';

export const shadows: Shadows = [
    'none', // 0: No shadow
    '0px 1px 2px rgba(0, 0, 0, 0.05)', // 1
    '0px 1px 3px rgba(0, 0, 0, 0.1)', // 2
    '0px 2px 4px rgba(0, 0, 0, 0.1)', // 3
    '0px 3px 5px rgba(0, 0, 0, 0.1)', // 4
    '0px 3px 6px rgba(0, 0, 0, 0.16)', // 5
    '0px 4px 8px rgba(0, 0, 0, 0.1)', // 6
    '0px 5px 10px rgba(0, 0, 0, 0.1)', // 7
    '0px 5px 15px rgba(0, 0, 0, 0.2)', // 8
    '0px 6px 20px rgba(0, 0, 0, 0.2)', // 9
    '0px 7px 20px rgba(0, 0, 0, 0.2)', // 10
    '0px 8px 24px rgba(0, 0, 0, 0.2)', // 11
    '0px 9px 28px rgba(0, 0, 0, 0.2)', // 12
    '0px 10px 32px rgba(0, 0, 0, 0.2)', // 13
    '0px 11px 36px rgba(0, 0, 0, 0.2)', // 14
    '0px 12px 40px rgba(0, 0, 0, 0.2)', // 15
    '0px 13px 44px rgba(0, 0, 0, 0.2)', // 16
    '0px 14px 48px rgba(0, 0, 0, 0.2)', // 17
    '0px 15px 52px rgba(0, 0, 0, 0.2)', // 18
    '0px 16px 56px rgba(0, 0, 0, 0.2)', // 19
    '0px 17px 60px rgba(0, 0, 0, 0.2)', // 20
    '0px 18px 64px rgba(0, 0, 0, 0.2)', // 21
    '0px 19px 68px rgba(0, 0, 0, 0.2)', // 22
    '0px 20px 72px rgba(0, 0, 0, 0.2)', // 23
    '0px 21px 76px rgba(0, 0, 0, 0.2)', // 24
];
