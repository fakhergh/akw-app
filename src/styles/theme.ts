import { createTheme } from '@mui/material/styles';

import { colorsScheme } from '@/styles/colors';
import { components } from '@/styles/components';
import { shadows } from '@/styles/shadows';

export const theme = createTheme({
    palette: {
        background: {
            default: colorsScheme.grey['100'],
            paper: colorsScheme.common.white,
        },
        ...colorsScheme,
    },
    spacing: (factor: number) => `${0.5 * factor}rem`,
    shadows,
    components,
});

export const drawerWidth = '22rem';
