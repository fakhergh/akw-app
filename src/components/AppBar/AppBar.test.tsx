import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { AppBar } from '@/components/AppBar/AppBar';
import { drawerWidth } from '@/styles/theme';

describe('<AppBar/>', () => {
    it('Should match snapshot', () => {
        const rendered = render(<AppBar drawerWidth={drawerWidth} />);

        expect(rendered).toMatchSnapshot();
    });
});
