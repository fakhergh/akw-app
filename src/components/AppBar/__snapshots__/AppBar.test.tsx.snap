// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AppBar/> > Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-lqmauj-MuiPaper-root-MuiAppBar-root"
        style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-70qvj9"
          >
            <button
              aria-label="btn-menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-cdtzx3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
            </button>
            <div
              class="MuiBox-root css-58wbk4"
            />
          </div>
          <div
            class="MuiBox-root css-i9gxme"
          />
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-lqmauj-MuiPaper-root-MuiAppBar-root"
      style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
      >
        <div
          class="MuiBox-root css-70qvj9"
        >
          <button
            aria-label="btn-menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-cdtzx3-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
              />
            </svg>
          </button>
          <div
            class="MuiBox-root css-58wbk4"
          />
        </div>
        <div
          class="MuiBox-root css-i9gxme"
        />
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
